src/main/java/com/github/osisoft/ocs_sample_library_preview/sds/StreamsClient.java

```diff
@@ -1305,4 +1316,43 @@ public void updateStreamType(String tenantId, String namespaceId, String streamI
             e.printStackTrace();
         }
     }
+
+    /***
+     * patches the access control of a stream
+     * 
+     * @param tenantId    tenant to work against
+     * @param namespaceId namespace to work against
+     * @param streamId    the stream to update the access control of
+     * @param patch       JsonArray
+     * @throws SdsError any error that occurs
+     */
+    public void patchAccessControl(String tenantId, String namespaceId, String streamId, JsonArray patch) throws SdsError {
+
+        try {
+            HttpClient httpClient = HttpClient.newHttpClient();
+
+            String url = baseUrl + getStreamPath.replace("{apiVersion}", apiVersion).replace("{tenantId}", tenantId)
+                    .replace("{namespaceId}", namespaceId).replace("{streamId}", streamId) + "/AccessControl";
+            URI uri = URI.create(url);
+            String body = mGson.toJson(patch);
+            HttpRequest request = baseClient.getRequest(uri).method("PATCH", BodyPublishers.ofString(body)).build();
+
+            HttpResponse<String> response = httpClient.send(request, BodyHandlers.ofString());
+            int httpResult = response.statusCode();
+            if (baseClient.isSuccessResponseCode(httpResult)) {
+            } else {
+                throw new Error("patch stream access control request failed");
+            }
+        } catch (Error error) {
+            throw error;
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (MalformedURLException mal) {
+            System.out.println("MalformedURLException");
+        } catch (IllegalStateException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
do we normally not close this list off with catch (Exception e)? The [hierarchy I'm looking at](https://vidvaan.com/java-exception-classes-hierarchy/) shows that everything either derives from Error or Exception, and we already have Error at the top.

Reply:
It looks like we had that catch in the `TypesClient` but not the `StreamsClient`. My gut says it's covered because it's caught inside the `baseClient.getRequest()` method, but I'm not sure. Won't do any harm to add it, so I added it to all the `StreamsClient` methods.